vta/hardware/chisel/src/test/scala/unittest/utils/RandomArray.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package unittest.util
+
+import scala.util.Random
+import scala.math.pow
+
+class RandomArray(val r: Random, val len: Int, val bits: Int) {
```

Comment:
What is the matter of having `Random` internally here? and the user providing it?

Reply:
I thought the user is passing the random object here so that they can maybe set the seed or something

Reply:
In that case, I think we should have the seed as an argument to the constructor? we can use a default value for the seed also. I think it might be harder to think on passing a `Random` object. Once you create a `RandomArray` object u should expect that all of this have been taken care of.

Reply:
I think we'll have it with no seed then, having a default seed kind of defeat the purpose of randomized test

Reply:
By default I meant like

`class MyClass (seed: Int = 0, ...)` so if user does not specify any seed then we will use zero or we can randomly use one.